jetty-server/src/main/java/org/eclipse/jetty/server/session/AbstractSessionDataStore.java

```diff
@@ -159,20 +228,101 @@ public void run()
             throw exception.get();
     }
 
+    @Override
+    public boolean exists(String id) throws Exception
+    {
+        final AtomicReference<Exception> exception = new AtomicReference<>();
+        final AtomicReference<Boolean> result = new AtomicReference<>();
+        Runnable r = new Runnable()
+        {
+            @Override
+            public void run()
+            {
+                try
+                {
+                    result.set(doExists(id));
+                }
+                catch (Exception e)
+                {
+                    exception.set(e);
+                }
+            }
+        };
+
+        _context.run(r);
+        if (exception.get() != null)
+            throw exception.get();
+        
+        return result.get();
+    }
+
     @Override
     public Set<String> getExpired(Set<String> candidates)
     {
         if (!isStarted())
             throw new IllegalStateException("Not started");
+        long now = System.currentTimeMillis();
         
+        // 1. always verify the set of candidates we've been given
+        //by the sessioncache
+        Set<String> expired = new HashSet<>();
+        Set<String> expiredCandidates = doGetExpired(candidates, now);
```

Comment:
Since the behaviour of  `doGetExpired` has changed a bit, is there a better name for it?